### PR TITLE
[SecurityBundle] resolve class name parameter inside AddSecurityVotersPass

### PR DESCRIPTION
**Pull request by pjarmalavicius**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #23733 
| License       | MIT

### CODE REVIEW

**Comment by chalasr** [comment]
Continuing discussion from the fixed ticket:

> Yes, it's the same as @chalasr suggests, but if we follow symfony best practices, then I would say that we also should write core to follow it. I mean class name parameters are not recommended and maybe one day it will be deprecated. Having check validity compiler pass doesn't depend on class parameter parsing and it also doesn't change what worked before

Resolving parameter in the existing pass could not break anything, only fix something that currently breaks (since `%parameter%` aren't valid class names).
Also I'm not fond of having a `TYPE_OPTIMIZE` pass which doesn't optimize actually.
If we want to fix this, I would say we should be consistent with the existing pass(es) doing it already, resolving parameters in the existing pass.

**Comment by chalasr** [comment]
Registering service class names was a convention which has been abandoned, we just stopped recommending to do so, we won't forbid it.

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/AddSecurityVotersPass.php [hunk 1]
```diff
@@ -42,12 +42,11 @@ public function process(ContainerBuilder $container)
...
 
         foreach ($voters as $voter) {
-            $class = $container->getDefinition((string) $voter)->getClass();
-
+            $definition = $container->getDefinition((string) $voter);
+            $class = $container->getParameterBag()->resolveValue($definition->getClass());
             if (!is_a($class, VoterInterface::class, true)) {
                 @trigger_error(sprintf('Using a security.voter tag on a class without implementing the %1$s is deprecated as of 3.4 and will be removed in 4.0. Implement the %1$s instead.', VoterInterface::class), E_USER_DEPRECATED);
             }
-
```
Comment (stof) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/AddSecurityVotersPass.php]: please don't remove the empty lines we have for readability.

**Review by stof — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @pjarmalavicius.

**Pull request closed without merging**